### PR TITLE
BET-4528

### PR DESCRIPTION
**Pull request by resmaa**
https://spectrocloud.atlassian.net/browse/BET-4528

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/625#" title="Author self-approved">resmaa</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
@resmaa: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | 846581f484d43356593a7b9819f82a4c4f710034 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/625/librarium_build/1506229066770944000) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@resmaa: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | 295280bbe2f0518931522870f6bc1c0418d42b5f | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/625/librarium_build/1506270717262434304) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@resmaa: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | 295280bbe2f0518931522870f6bc1c0418d42b5f | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/625/librarium_build/1506474688069505024) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: packages/docs/content/04-clusters/3-cluster-management/9-cluster-rbac.md [hunk 1]
```diff
@@ -91,3 +91,119 @@ Configure settings as described above.
...
 * Use ClusterRole and RoleBinding to scope security to several or all namespaces.
 * Use ClusterRole and ClusterRoleBinding to scope security to all namespaces OR cluster-scoped resources.
+
+# Use RBAC-OIDC in your Public Cloud 
+
+
+This section explains the RBAC OIDC configuration to be done for all the public cloud except [Azure-AKS](/clusters/new-clusters/aks/#configureazureactivedirectory) and [EKS](/integrations/oidc-eks/) clusters.
+
+```json
+extraArgs:-
```
Comment (codrin-iftimie) [inline packages/docs/content/04-clusters/3-cluster-management/9-cluster-rbac.md:101]: this `-` might be extra

File: packages/docs/content/04-clusters/3-cluster-management/9-cluster-rbac.md [hunk 2]
```diff
@@ -91,3 +91,119 @@ Configure settings as described above.
 * Use Role and a RoleBinding to scope security to a single namespace.
 * Use ClusterRole and RoleBinding to scope security to several or all namespaces.
 * Use ClusterRole and ClusterRoleBinding to scope security to all namespaces OR cluster-scoped resources.
+
+# Use RBAC-OIDC in your Public Cloud 
+
+
+This section explains the RBAC OIDC configuration to be done for all the public cloud except [Azure-AKS](/clusters/new-clusters/aks/#configureazureactivedirectory) and [EKS](/integrations/oidc-eks/) clusters.
+
+```json
```
Comment (codrin-iftimie) [inline packages/docs/content/04-clusters/3-cluster-management/9-cluster-rbac.md:100]: yaml

File: packages/docs/content/04-clusters/3-cluster-management/9-cluster-rbac.md [hunk 3]
```diff
@@ -91,3 +91,119 @@ Configure settings as described above.
...
+extraArgs:-
+      oidc-issuer-url: "provider URL"
+      oidc-client-id: "client-id"
+      oidc-groups-claim: "groups"
+      oidc-username-claim: "email"
+```
+## Client configuration to add OIDC based authentication flags in kubeconfig
+
+```json
```
Comment (codrin-iftimie) [inline packages/docs/content/04-clusters/3-cluster-management/9-cluster-rbac.md:109]: yaml

File: packages/docs/content/04-clusters/3-cluster-management/9-cluster-rbac.md [hunk 4]
```diff
@@ -91,3 +91,119 @@ Configure settings as described above.
...
+
+
+This section explains the RBAC OIDC configuration to be done for all the public cloud except [Azure-AKS](/clusters/new-clusters/aks/#configureazureactivedirectory) and [EKS](/integrations/oidc-eks/) clusters.
+
+```json
+extraArgs:-
+      oidc-issuer-url: "provider URL"
+      oidc-client-id: "client-id"
+      oidc-groups-claim: "groups"
+      oidc-username-claim: "email"
```
Comment (codrin-iftimie) [inline packages/docs/content/04-clusters/3-cluster-management/9-cluster-rbac.md:105]: lets use the same indentations as the rest (2 spaces)

File: packages/docs/content/04-clusters/3-cluster-management/9-cluster-rbac.md [hunk 5]
```diff
@@ -91,3 +91,119 @@ Configure settings as described above.
...
+```json
+clientConfig:
+  oidc-issuer-url: "https://dev-13018903.okta.com"
+  oidc-client-id: "0oa1bqubeezLzCAPP5d7"
+  oidc-client-secret: "378VO445Xoq_wGbV0wD-mlWRrPWu8dlWdPK4fKJ7"
+  oidc-extra-scope: profile,email,openid
+```
+## An Example Scenario: GCP- RBAC Configuration
+
+```json
```
Comment (codrin-iftimie) [inline packages/docs/content/04-clusters/3-cluster-management/9-cluster-rbac.md:118]: yaml

**Review by codrin-iftimie — CHANGES_REQUESTED** [review]

**Review by codrin-iftimie — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/625#pullrequestreview-920036053" title="Approved">codrin-iftimie</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/625#" title="Author self-approved">resmaa</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**